modules/farm_fd2/src/entrypoints/cover_crop_seeding/lib.submitOptional.unit.cy.js

```diff
@@ -1,121 +1,142 @@
-      expect(res.seedIncorporationSpeedQuantity).to.exist;
-      expect(res.seedIncorporationAreaQuantity).to.exist;
-      expect(res.seedIncorporationActivityLog).to.exist;
+  // No application equipment
+  describe('When no seed application equipment is provided', () => {
+    let results = null;
+    const form = {
+      date: '1950-01-02',
+      crops: ['BEAN'],
+      location: 'ALF',
+      beds: [],
+      areaSeeden: 50,
+      seedApplicationEquipment: [], // Empty for this test
+      seedIncorporationEquipment: ['Rake'],
+      seedApplicationDepth: 0,
+      seedApplicationSpeed: 0,
+      seedApplicationPasses: 1,
+      seedIncorporationDepth: 8,
+      seedIncorporationSpeed: 3,
+      seedIncorporationPasses: 1,
+      winterKill: false,
+      winterKillDate: null,
+      comment: 'A comment',
+    };
+
+    before(() => {
+      cy.wrap(lib.submitForm(form), { timeout: 30000 }).then((res) => {
+        results = res;
+      });
+    });
+
+    it('should NOT create seed application records', () => {
+      expect(results).to.not.have.property('seedApplicationActivityLog0');
+      expect(results).to.not.have.property('seedApplicationDepthQuantity0');
+    });
+
+    it('should still create seed incorporation records with correct details', () => {
+      const log = results.seedIncorporationActivityLog0;
+      expect(log.type).to.equal('log--activity');
+      expect(log.relationships.equipment[0].id).to.equal(
+        equipmentMap.get(form.seedIncorporationEquipment[0]).id
+      );
     });
   });
 
-  it('Check that seed incorporation logs are not created if no seedIncorporationEquipment is provided', () => {
-    cy.wrap(
-      lib.submitForm({
-        date: '1950-01-02',
-        crops: ['BEAN', 'CARROT'],
-        location: 'ALF',
-        beds: ['ALF-1', 'ALF-3'],
-        areaSeeded: 50,
-        seedApplicationEquipment: ['Tractor'],
-        seedIncorporationEquipment: [],
-        seedApplicationDepth: 6,
-        seedApplicationSpeed: 5,
-        seedIncorporationDepth: 0,
-        seedIncorporationSpeed: 0,
-        winterKill: true,
-        winterKillDate: '1950-12-31',
-        comment: 'A comment',
-      }),
-      {
-        timeout: 10000,
-      }
-    ).then((res) => {
-      expect(res.plantAsset).to.exist;
-      expect(res.areaSeededQuantity).to.exist;
-      expect(res.seedingLog).to.exist;
-      expect(res.winterKillLog).to.exist;
-      expect(res.seedApplicationDepthQuantity).to.exist;
-      expect(res.seedApplicationSpeedQuantity).to.exist;
-      expect(res.seedApplicationAreaQuantity).to.exist;
-      expect(res.seedApplicationActivityLog).to.exist;
-      expect(res.seedIncorporationDepthQuantity).to.be.null;
-      expect(res.seedIncorporationSpeedQuantity).to.be.null;
-      expect(res.seedIncorporationAreaQuantity).to.be.null;
-      expect(res.seedIncorporationActivityLog).to.be.null;
+  // No equipment at all
+  describe('Submission without equipment', () => {
```

Comment:
Seems like this test should be checking for no seed incorporation equipment instead of no equipment at all.